lib/galaxy/datatypes/mothur.py

```diff
@@ -0,0 +1,937 @@
+"""
+Mothur Metagenomics Datatypes
+"""
+import logging
+import sys
+import re
+from galaxy.datatypes.sniff import get_headers
+from galaxy.datatypes.metadata import MetadataElement
+from galaxy.datatypes.data import Text
+from galaxy.datatypes.tabular import Tabular
+
+log = logging.getLogger(__name__)
+
+
+class Otu(Text):
+    file_ext = 'mothur.otu'
+    MetadataElement(name="columns", default=0, desc="Number of columns", readonly=True, visible=True, no_value=0)
+    MetadataElement(name="labels", default=[], desc="Label Names", readonly=True, visible=True, no_value=[])
+
+    def __init__(self, **kwd):
+        Text.__init__(self, **kwd)
+
+    def set_meta(self, dataset, overwrite=True, **kwd):
+        if dataset.has_data():
+            label_names = set()
+            ncols = 0
+            data_lines = 0
+            comment_lines = 0
+
+            headers = get_headers(dataset.file_name, sep='\t', count=-1)
+            for line in headers:
+                if len(line) >= 2:
```

Comment:
I was wondering whether for OTU files a comment line doesn't start with `@`. If so, I think it would be better to use `startswith('@')` which is consistent with the other code